lib/gtm.ts

```diff
@@ -0,0 +1,5 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-condition */
+export const initializeGTM = () => {
+  window.dataLayer = window.dataLayer || [];
+  window.dataLayer.push({ "gtm.start": new Date().getTime(), event: "gtm.js" });
```

Comment:
Is there any reference on how this could delay the initializations and improve the performance?

Reply:
@zainfathoni Since this is called on the `useEffect()` hook, it will follow the hooks lifecycle. No idea on how this will be any faster, but at least this fixes the multiple `<script />` tags issue, since for some reason only inline scripts are affected.

However, this will probably be fixed in a future update to Next.js. (https://github.com/vercel/next.js/pull/27218/files)